.github/workflows/publish-site.yml

```diff
@@ -55,8 +53,8 @@ jobs:
         id: site-changes
         if: steps.check-ghpages.outputs.exists == 'true'
         run: |
-          git diff --stat origin/gh-pages master:doc/site
-          git diff --quiet origin/gh-pages master:doc/site && (echo "modified=false" > $GITHUB_OUTPUT) || (echo "modified=true" > $GITHUB_OUTPUT)
+          git diff --stat origin/gh-pages ${{ github.ref_name }}:doc/site
```

Comment:
@badosu pls see changes made here. `github.ref_name` should be the branch that triggered the merge. This allows testing of this workflow from an alternative branch during development.

The removal of `ref:` from `actions/checkout` also will default to the action branch.

In general I expect this will still run on master when executed by `cron`.

Reply:
That said, it might make sense to run this job when master changes rather than on a timer? I also noticed that it's running on my fork (and presumably every other fork), and just failing every time.

Reply:
I'd rather have it on a timer simply due to the fact it's not critical docs are updated more frequently than daily, but we have more updates than once a day. But this is  my opinion.